ngi_reports/utils/entities.py

```diff
@@ -719,73 +685,120 @@ def populate(self, log, organism_names, **kwargs):
-
-        # Get all samples from flow cell that belong to the project
-        fc_samples = []
-        for fc_sample in fcObj.sample_sheet_data:
-            if fc_sample.get("Sample_Name").split("_")[0] == self.ngi_id:
-                fc_samples.append(fc_sample.get("Sample_Name"))
-
-        # Go through all samples in project to identify their prep_ID (only if they are on the flowcell)
-        for sample_ID in self.samples:
-            for prep_ID in self.samples.get(sample_ID).preps:
-                sample_preps = self.samples.get(sample_ID).preps
-                if fcObj.name in sample_preps.get(prep_ID).seq_fc:
-                    preps_samples_on_fc.append([sample_ID, prep_ID])
+        for fcObj in self.flowcells.values():
+            preps_samples_on_fc = []
+            additional_samples = []
+
+            # Get all samples from flow cell that belong to the project
+            fc_samples = []
+            for fc_sample in fcObj.sample_sheet_data:
+                if fc_sample.get("Sample_Name").split("_")[0] == self.ngi_id:
+                    fc_samples.append(fc_sample.get("Sample_Name"))
+
+            # Go through all samples in project to identify their prep_ID (only if they are on the flowcell)
+            for sample_ID in self.samples:
+                sampleObj = self.samples.get(sample_ID)
+                for prep_ID in sampleObj.preps:
+                    prepObj = sampleObj.preps.get(prep_ID)
+                    if prepObj.barcode != "NA" and prepObj.qc_status != "NA":
+                        prepObj.seq_fc = []
+                        if (
+                            not sampleObj.sample_info.get("library_prep")
+                            .get(prep_ID)
+                            .get("sequenced_fc")
+                        ):
+                            log.error(
+                                "Sequenced flowcell not defined for the project. "
+                                'Run ngi_pipelines without the "-b" flag and amend the report manually.'
+                            )
+                            sys.exit("Stopping execution...")
+                        for fc in (
+                            sampleObj.sample_info.get("library_prep")
+                            .get(prep_ID)
+                            .get("sequenced_fc")
+                        ):
+                            prepObj.seq_fc.append(fc.split("_")[-1])
+                    if fcObj.name in prepObj.seq_fc:
+                        preps_samples_on_fc.append([sample_ID, prep_ID])
+                    else:
+                        continue
+
+                list_of_barcodes = [prep.barcode for prep in sampleObj.preps.values()]
+                if len(list_of_barcodes) >= 1:
+                    list_of_flowcells = [
+                        prep.seq_fc for prep in sampleObj.preps.values()
+                    ]
+                    if len(list_of_flowcells) != len(
+                        list_of_flowcells
```

Comment:
`len(list_of_flowcells) != len(list_of_flowcells):`
what is being checked here?

Reply:
Whoops 😂 
It used to be `len(list_of_flowcells) != len(dict.fromkeys(list_of_flowcells))` and I thought I'd simplify it but I guess I got a bit lost. This check was already a bit broken I think, maybe @FranBonath can shed some light on how it's supposed to work? 😄

Reply:
I'll remove it for now since it was already broken. The -b option isn't accredited and needs to be looked over more anyway.